app/controllers/movie_preference_controller.py

```diff
@@ -0,0 +1,22 @@
+from flask import abort, jsonify, request
+
+from app import db
+from app.models import MoviePreference, User
+from app.util.session_util import get_current_session_user
+
+
+class MoviePreferenceController():
+    def handle(self, user_id):
+        user = get_current_session_user()
+        if user is not User.query.get(user_id):
+            abort(403)
+
+        data = request.get_json()
+        external_movie_id = data.get('externalMovieId', None)
+        if not external_movie_id:
+            abort(400)
+
+        mp = MoviePreference(user, external_movie_id)
+        db.session.commit()
```

Comment:
If this line fails because they already have that preference, what response would the user get?

Reply:
It wouldn't 😬I have a follow-up branch for that, should I just add that in now? It's pretty chunky (migration, error handling, etc)

Reply:
Nah, save for a followup is fine